tests/driver_lm75/main.c

```diff
@@ -0,0 +1,207 @@
+
+#include "board.h"
+#include "lm75.h"
+#include "lm75_regs.h"
+#include "lm75_params.h"
+#include "periph/i2c.h"
+#include "periph/gpio.h"
+#include "xtimer.h"
+#include "log.h"
+#include "kernel_defines.h"
+#include <stdio.h>
+#include <stdint.h>
+
+#define I2C_BUS       dev->lm75_params.i2c_bus
+#define I2C_ADDR      dev->lm75_params.i2c_addr
+
+/* prints currently set OS and HYST temperatures */
+int lm75_check_set_temperature_limits(lm75_t *dev) {
+
+    int t_os, t_hyst;
+    /* get already set OS and HYST values */
+    lm75_get_os_temp(dev, &t_os);
+    printf("Set OS temp is %d.%dºC\n",   t_os / dev->lm75_params.res->os_mult, \
+                                         t_os % dev->lm75_params.res->os_mult);
+
+    lm75_get_hyst_temp(dev, &t_hyst);
+    printf("Set HYST temp is %d.%dºC\n", t_hyst / dev->lm75_params.res->os_mult, \
+                                         t_hyst % dev->lm75_params.res->os_mult);
+    return 0;
+}
+
+/* This function prints the current temperature */
+int lm75_print_temperature(lm75_t *dev) {
+
+    int temp;
+    lm75_get_temperature(dev, &temp);
+
+    /* this is required not to discard the leading zero and
+    * output a false result in the case the temperature is x.0625 */
+    if (dev->lm75_params.res->temp_mult > 1000 \
+         && temp % dev->lm75_params.res->temp_mult == 625) {
+
+        printf("%d.0%dºC\n", temp / dev->lm75_params.res->temp_mult, \
+                             temp % dev->lm75_params.res->temp_mult);
+    }
+    else {
+        printf("%d.%dºC\n", temp / dev->lm75_params.res->temp_mult, \
+                            temp % dev->lm75_params.res->temp_mult);
+    }
+
+    return 0;
+}
+
+/* This function makes the device measure temperatures in a strictly discrete way
+ * at a user definable rate, as opposed to performing continuous measurements
+ * at the device's sampling rate.
+ * It allow the device to stay in shutdown mode for the most part, therefore consuming less power.
+ * In the tmp1075 sensor this is done automatically using the one shot mode feature.
+ * In the LM75A sensor this is done manually by switching the device to and from shutdown mode. */
+int lm75_low_power_mode(lm75_t *dev, uint16_t interval) {
+
+    if (i2c_acquire(I2C_BUS) != 0) {
+        return LM75_ERROR_I2C;
+    }
+
+    uint8_t config;
+    if (i2c_read_reg(I2C_BUS, I2C_ADDR, LM75_CONF_REG, &config, 0) != 0) {
+        i2c_release(I2C_BUS);
+        return LM75_ERROR_I2C;
+    }
+    i2c_release(I2C_BUS);
+
+    config &= 0b00000001;
+    if (config != 1) {
+        puts("not in shutdown mode");
+        return LM75_ERROR;
+    }
+
+    if (IS_USED(MODULE_TMP1075)) {
+        if (tmp1075_one_shot(dev) != 0) {
+            return LM75_ERROR;
+        }
+        xtimer_msleep(interval);
+    }
+    else {
+        lm75_poweron(dev);
+        /* since the temp is only updated every 100ms in the lm75 sensor
+         * this is required to ensure the temp register updates for followup readings */
```

Comment:
This requirement should be enforced by the driver (e.g. returning `-EAGAIN`) not by user code.